build.zig

```diff
@@ -1155,31 +1195,42 @@ const libImGuiOptions = struct {
     optimize: std.builtin.OptimizeMode,
     use_ubsan: bool = false,
     use_tsan: bool = false,
+    lib_sokol: *CompileStep,
+    emsdk: ?*Build.Dependency = null,
+    version: []const u8,
 };
 const imguiVersion = enum {
     default,
     docking,
 };
 fn buildImgui(b: *Build, options: libImGuiOptions) !*CompileStep {
-
-    // get the imgui source code
-    const imguiver_path = switch (b.option(
-        imguiVersion,
-        "imgui-version",
-        "Select ImGui version to use",
-    ) orelse imguiVersion.default) {
-        .default => "src",
-        .docking => "src-docking",
-    };
-
     const libimgui = b.addStaticLibrary(.{
         .name = "imgui",
         .target = options.target,
         .optimize = options.optimize,
     });
+
     libimgui.root_module.sanitize_c = options.use_ubsan;
     libimgui.root_module.sanitize_thread = options.use_tsan;
 
+    if (options.target.result.isWasm()) {
```

Comment:
Here, I've tried avoiding redundancies in calling emsdk using the below workaround. However, I still get the same error!
```zig
for(options.lib_sokol.root_module.include_dirs.items) |dir| {
        try libimgui.root_module.include_dirs.append(b.allocator,dir);
    }
```